core/storage/file/gae_models_test.py

```diff
@@ -0,0 +1,229 @@
+        file_metadata_model2 = file_models.FileMetadataModel.create(
+            'exp_id1', 'path/to/file2.png')
+        file_metadata_model2.commit(feconf.SYSTEM_COMMITTER_ID, [])
+        file_metadata_model2.delete(
+            feconf.SYSTEM_COMMITTER_ID, 'Delete second file model')
+        file_metadata_model_list = (
+            file_models.FileMetadataModel.get_undeleted())
+        self.assertEqual(len(file_metadata_model_list), 1)
+        self.assertEqual(
+            file_metadata_model1, file_metadata_model_list[0])
+
+    def test_create_creates_model_with_correct_id_and_deleted_status_as_false(
+            self):
+        file_metadata_model1 = file_models.FileMetadataModel.create(
+            'exp_id1', 'path/to/file1.png')
+        self.assertEqual(file_metadata_model1.deleted, False)
+        self.assertEqual(
+            file_metadata_model1.id, '/exp_id1/path/to/file1.png')
+
+        file_metadata_model2 = file_models.FileMetadataModel.create(
+            'exp_id1', '/exp_id1/path/to/file2.png')
+        self.assertEqual(file_metadata_model2.deleted, False)
+        self.assertEqual(
+            file_metadata_model2.id, '/exp_id1/path/to/file2.png')
+
+    def test_get_model_with_model_present_returns_the_correct_model(self):
+        file_metadata_model = file_models.FileMetadataModel.create(
+            'exp_id1', 'path/to/file1.png')
+        file_metadata_model.commit(feconf.SYSTEM_COMMITTER_ID, [])
+        retrieved_model = file_models.FileMetadataModel.get_model(
+            'exp_id1', 'path/to/file1.png')
+
+        self.assertEqual(retrieved_model, file_metadata_model)
+
+    def test_get_model_non_strict_with_no_model_present_returns_none(self):
+        retrieved_model = file_models.FileMetadataModel.get_model(
+            'exp_id1', 'path/to/file2.png')
+        self.assertIsNone(retrieved_model)
+
+    def test_get_model_strict_with_no_model_present_raises_error(self):
+        with self.assertRaisesRegexp(
+            base_models.BaseModel.EntityNotFoundError, (
+                'Entity for class FileMetadataModel '
+                'with id /exp_id1/path/to/file2.png not found')):
+            file_models.FileMetadataModel.get_model(
+                'exp_id1', 'path/to/file2.png', True)
+
+    def test_get_version_with_version_present_returns_correct_model(self):
+        file_metadata_model = file_models.FileMetadataModel.create(
+            'exp_id1', 'path/to/file1.png')
+        file_metadata_model.commit(feconf.SYSTEM_COMMITTER_ID, [])
+
+        retrieved_model = file_models.FileMetadataModel.get_version(
+            'exp_id1', 'path/to/file1.png', 1)
+        self.assertEqual(file_metadata_model.key, retrieved_model.key)
+
+        file_metadata_model.commit(feconf.SYSTEM_COMMITTER_ID, [])
+        retrieved_model = file_models.FileMetadataModel.get_version(
+            'exp_id1', 'path/to/file1.png', 2)
+        self.assertEqual(file_metadata_model.key, retrieved_model.key)
```

Comment:
I see the benefit in verifying by key, but is there anything property-wise that we can verify to ensure this is correct? E.g. relying on key can be a bit precarious since it may not always be necessary to include the version in the key (unless that's a specific contract of the underlying model that I'm not aware of). Instead, it might be clearer to verify any other public properties of the model which will change with the version to ensure it's the correct one in addition to the key or ID.